File: strategy/src/main/scala/com/nearsoft/workshop/strategy/scala/character/King.scala
package com.nearsoft.workshop.strategy.scala.character

import org.slf4j.LoggerFactory

class King extends Character {

  val log = LoggerFactory.getLogger(classOf[King])

  override def fight = {
    log.debug(s"I'M THE KING, and I'm ${useWeapon}")
  }
}

File: strategy/src/main/java/com/nearsoft/workshop/strategy/character/King.java
package com.nearsoft.workshop.strategy.character;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class King extends Character {

    private final Logger log = LoggerFactory.getLogger(this.getClass());

    @Override
    public void fight() {
        log.debug(String.format("I'M THE KING, and I'm %s", weaponBehaviour.useWeapon()));
    }
}
